File: showcase/guards_lua/scripts/Beacon.lua
---@uuid 7c259fd2-fdb9-453b-a7ef-19cdd85428cc
---@class Beacon : NodeScript
Beacon = script_class()

function Beacon:on_update()
    local pos = self.node.global_position
    table.insert(GlobalScript:get("Game").beacons, pos)
    print("beacon registered: " .. tostring(self.node));
    self.node:destroy()
end

File: showcase/guards_cs/scripts/Beacon.cs
[Uuid("7c259fd2-fdb9-453b-a7ef-19cdd85428cc")]
public class Beacon : NodeScript
{
    protected override void OnUpdate(float dt)
    {
        GlobalScript.Get<Game>().Beacons.Add(Node.GlobalPosition);

        Log.Info($"Beacon registered: {Node}");

        Node.Destroy();
    }
}